l1/eth_subscriber.go

```diff
@@ -41,12 +41,30 @@ func NewEthSubscriber(ethClientAddress string, coreContractAddress common.Addres
 	}
 	return &EthSubscriber{
 		ethClient: ethClient,
-		client:    client,
+		client:    ethClient.Client(),
 		filterer:  filterer,
 		listener:  SelectiveListener{},
 	}, nil
 }
 
+//go:generate mockgen -destination=../mocks/mock_ethclient.go -package=mocks  github.com/NethermindEth/juno/l1 EthClient
+type EthClient interface {
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+}
+
+func NewETHClient(ethClientAddress string) (EthClient, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	// TODO replace with our own client once we have one.
+	// Geth pulls in a lot of dependencies that we don't use.
+	client, err := rpc.DialContext(ctx, ethClientAddress)
+	if err != nil {
+		return nil, err
+	}
+	ethClient := ethclient.NewClient(client)
+	return ethClient, nil
+}
+
```

Comment:
The RPC handler only needs access to the `TransactionReceipt` method and I thought it might have been neater to constrain `ethClient` using a new interface. But I updated the code to remove the EthClient interface, and grant the RPC handler access to `Subscriber` instead (no need to give it access to everything in `EthSubscriber` imo)

Reply:
I think you are right we should not give access to more than necessary but it is not the correct place to do this. 

You can achieve the same behaviour by moving the `EthClient` to the rpc's package and in the node package pass the subscriber since the subscriber will implement the `EthClient` interface.